src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs

```diff
@@ -110,6 +121,62 @@ internal static ImmutableArray<MetadataReference> MakeAssemblyReferences(this Im
             return referencesBuilder.ToImmutableAndFree();
         }
 
+        /// <summary>
+        /// Add aliases to any strong-named duplicate assemblies that differ
+        /// by version, preferring the version referenced from the target module.
+        /// </summary>
+        private static void AddAliasesToDuplicateAssemblies(
+            ArrayBuilder<MetadataReference> references,
+            ArrayBuilder<AssemblyIdentity> identities,
+            AssemblyIdentityComparer identityComparer,
+            ArrayBuilder<AssemblyIdentity> referencedModules)
+        {
+            Debug.Assert(references.Count == identities.Count);
+
+            // Dictionary to ensure exactly one unique alias
+            // is added to each AssemblyIdentity.
+            var duplicateIdentities = PooledDictionary<AssemblyIdentity, string>.GetInstance();
+
+            // Find duplicate pairs.
+            int n = identities.Count;
+            foreach (var referencedModule in referencedModules)
+            {
+                if (!referencedModule.IsStrongName)
+                {
+                    continue;
+                }
+                for (int i = 0; i < n; i++)
+                {
+                    var identity = identities[i];
+                    var compareResult = identityComparer.Compare(referencedModule, identity);
+                    // Only need to handle assemblies that differ by version since
+                    // ReferenceManager will drop equivalent assemblies.
+                    if (compareResult != AssemblyIdentityComparer.ComparisonResult.EquivalentIgnoringVersion)
+                    {
+                        continue;
+                    }
+                    duplicateIdentities[identity] = Guid.NewGuid().ToString();
+                }
+            }
+
+            Debug.Assert(references.All(r => r.Properties.Aliases.IsEmpty));
```

Comment:
Why is this true?  Does this code run before we re-constitute extern aliases from custom debug info?